ddtrace/internal/writer.py

```diff
@@ -64,19 +67,41 @@ def flush_queue(self):
             log.error('error while filtering traces: {0}'.format(err))
             return
 
-        if not traces:
-            return
+        traces_flushed = len(traces)
+        traces_flushed_error = 0
 
         # If we have data, let's try to send it.
         traces_responses = self.api.send_traces(traces)
         for response in traces_responses:
             if isinstance(response, Exception) or response.status >= 400:
                 self._log_error_status(response)
+                traces_flushed_error += 1
             elif self._priority_sampler:
                 result_traces_json = response.get_json()
                 if result_traces_json and 'rate_by_service' in result_traces_json:
                     self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
 
+        # Dump statistics
+        # NOTE: Do not use the buffering of dogstatsd as it's not thread-safe
+        # https://github.com/DataDog/datadogpy/issues/439
+        if self.dogstatsd:
+            self.dogstatsd.increment('datadog.tracer.traces.api.requests', len(traces_responses))
+            for status, grouped_responses in itertools.groupby(
+                    sorted(traces_responses, key=lambda r: r.status),
+                    key=lambda r: r.status):
+                self.dogstatsd.increment('datadog.tracer.traces.api.responses.count',
```

Comment:
```suggestion
                self.dogstatsd.increment('datadog.tracer.traces.api.responses',
```

Reply:
we can probably just drop the `traces.` part of these `api` ones.

the `tracer.traces.*` ones can represent internal enqueueing/filtering/etc

and `tracer.api.*` can be specifically for the API calls to the agent.


no strong opinion here, just a thought.

Reply:
Agreed.